### ISSUE TITLE
Wildcard support for code-splitting

### ISSUE THREAD

**Issue by myndzi** [webpack/webpack#1592]
To my current understanding, if you want to split off vendor code from application code, you have to provide an explicit list of the modules that you want in the vendor code. The trouble with this is that you must maintain an explicit list and keep it in sync with changes to your application. Webpack already discerns the information about which files are required via an entry point, so it makes quite a bit of sense to be able to specify a glob or regex to match against the bundled modules to define which ones are split into another chunk.

In other words, something like this would be very convenient, and is easy for webpack to provide but hard for a consumer to implement:

```
{
    entry: {
        app: __dirname + '/client/app.js',
        vendor: __dirname + '/node_modules/**'
    }
}
````

**Comment by taurose** [comment]
I believe this is what you want: [link](https://stackoverflow.com/questions/26603499/simple-solution-to-share-modules-loaded-via-npm-across-multiple-browserify-or-we/29087883#29087883) (using `minChunks` function parameter for `CommonsChunkPlugin`).

I agree that it would be nice to have a more straightforward way of setting that up. Using glob/regex for entry point doesn't seem right though; I'd expect that to just bundle all the matches it can find on the file system like `require.context` does.

**Comment by myndzi** [comment]
Yeah, that solves the problem in a reasonably succinct way. I don't feel like it's on uncommon use case though, so clear support or documentation would be a bonus. Thanks!
